src/inmanta/agent/forking_executor.py

```diff
@@ -682,3 +752,41 @@ async def stop_for_agent(self, agent_name: str) -> list[MPExecutor]:
         children = [self.executor_map[child_id] for child_id in children_ids]
         await asyncio.gather(*(child.stop() for child in children))
         return children
+
+    def start_periodic_executor_cleanup(self) -> None:
+        async def cleanup_inactive_executors() -> None:
```

Comment:
If the manager is stopped or stopping, you have to break out of this loop as well (or cancel it or both)